### PR TITLE
Revisit the default Scheduled Jobs. Adjust the frequency…

### PR DESCRIPTION
**Pull request by agileware-justin**
…. Remove comments from API parameters which cause the Job to not execute. Update description with parameters comments

Overview
----------------------------------------
CiviCRM creates default Scheduled Jobs which will not execute when enabled because the parameters include comments and/or have an invalid format. This is a common problem with new CiviCRM sites and even old ones which have been installed and the Scheduled Jobs never really checked thoroughly.

People will typically assume Scheduled Jobs are set up to work and just enable them. It happens alot.

The default frequency of some of the Scheduled Jobs is also too frequent and this PR adjusts that schedule to something more akin to what is usable in the typical shared hosting environment with limited resources.

Have added the following new jobs:

- **Group Cache Flush** - as that has been a long time recommendation.
- **Update Individual Postal Greeting** - required if this greeting is to be updated (which I had always assumed was done already)
- **Update Individual Addressee** - required if this greeting is to be updated
- **Dedupe Contacts** - as this is something that would be really useful to have set up from Day 1!

And re-named this job: Update Individual Email Greeting

Before
----------------------------------------
Some Scheduled Jobs do not work by default.
Parameters include comments.
Frequency of some Scheduled Jobs is too frequent.
Group Cache Flush job does not exist.

After
----------------------------------------
Scheduled Jobs will work by default if enabled.
Comments are moved to the Job Description.
Frequency of some Scheduled Jobs has been adjusted.
Some new Scheduled Jobs have been added, see above.

Technical Details
----------------------------------------

Comments
----------------------------------------
Happy to discuss and listen to any feedback improvements or other ways to ensure that Scheduled Jobs work out of the box.

Agileware Ref: CIVICRM-2189

### CODE REVIEW

**Comment by eileenmcnaughton** [comment]
@agileware-justin I think one of the descriptions must be too long - test fails

PR@2/BKPROF/dfl/SUITES/phpunit-api3/label/bknix-tmp/junit api_v3_AllTests
## Started: Tue Nov  7 06:27:35 PM PST 2023
######################################################################
RUN:[[./scripts/phpunit --tap --log-junit '/home/homer/workspace/CiviCRM-Core-Matrix-PR@2/BKPROF/dfl/SUITES/phpunit-api3/label/bknix-tmp/junit/api_v3_AllTests.xml' '--exclude-group' 'ornery' 'api_v3_AllTests']]
Installing build4test_nw8hk schema
PHPUnit 9.6.5 by Sebastian Bergmann and contributors.

Error in bootstrap script: PDOException:
**SQLSTATE[22001]: String data, right truncated: 1406 Data too long for column 'description' at row 6**
#0 /home/homer/buildkit/build/build-4/web/sites/all/modules/civicrm/Civi/Test.php(243): PDO->query('INSERT INTO civ...')
#1 /home/homer/buildkit/build/build-4/web/sites/all/modules/civicrm/Civi/Test/Data.php(32): Civi\Test::execute('INSERT INTO civ...')
#2 /home/homer/buildkit/build/build-4/web/sites/all/modules/civicrm/Civi/Test.php(40): Civi\Test\Data->Civi\Test\{closure}()
#3 /home/homer/buildkit/build/build-4/web/sites/all/modules/civicrm/Civi/Test/Data.php(46): Civi\Test::asPreInstall(Object(Closure))
#4 /home/homer/buildkit/build/build-4/web/sites/all/modules/civicrm/Civi/Test.php(126): Civi\Test\Data->populate()
#5 [internal function]: Civi\Test::Civi\{closure}(Object(Civi\Test\CiviEnvBuilder))
#6 /home/homer/buildkit/build/build-4/web/sites/all/modules/civicrm/Civi/Test/CiviEnvBuilder/CallbackStep.php(27): call_user_func(Object(Closure), Object(Civi\Test\CiviEnvBuilder))
#7 /home/homer/buildkit/build/build-4/web/sites/all/modules/civicrm/Civi/Test/CiviEnvBuilder.php(196): Civi\Test\CiviEnvBuilder\CallbackStep->run(Object(Civi\Test\CiviEnvBuilder))

**Comment by agileware-justin** [comment]
Ta @eileenmcnaughton have reworked the descriptions.

File: sql/civicrm_data/civicrm_job.sqldata.php [hunk 1]
```diff
@@ -41,42 +40,45 @@
...
       'api_entity' => 'Job',
       'api_action' => 'geocode',
-      'parameters' => 'geocoding=[1 or 0] required
-parse=[1 or 0] required
-start=[contact ID] optional-begin with this contact ID
-end=[contact ID] optional-process contacts with IDs less than this
-throttle=[1 or 0] optional-1 adds five second sleep',
+      'parameters' => 'geocoding=1
+      parse=0
+      throttle=0',
```
Comment (colemanw) [inline sql/civicrm_data/civicrm_job.sqldata.php:60]: I think the lack of leading spaces was actually intentional here as each param is supposed to go at the start of a newline.

File: sql/civicrm_data/civicrm_job.sqldata.php [hunk 2]
```diff
@@ -41,42 +40,45 @@
...
     ],
     [
       'run_frequency' => 'Daily',
       'name' => 'Update Greetings and Addressees',
-      'description' => 'Goes through contact records and updates email and postal greetings, or addressee value',
+      'description' => 'Updates email and postal greetings, or addressee value. Use parameters: ct=[Individual or Household or Organization] required
+gt=[email_greeting or postal_greeting or addressee] required
+force=[0 or 1]
+limit=[Number] - Limit contacts to update',
```
Comment (colemanw) [inline sql/civicrm_data/civicrm_job.sqldata.php:68]: However, in the description the newlines don't make much sense.

File: sql/civicrm_data/civicrm_job.sqldata.php (hunk 1 above)
Comment (mattwire) [reply sql/civicrm_data/civicrm_job.sqldata.php:60]: Yes that's correct.

**Comment by eileenmcnaughton** [comment]
test fail doesn't look related ( - test this please )

Overall this looks like nice work - so keen to see this merged once the above feedback is worked through

**Comment by agileware-justin** [comment]
PR updated. I think it makes sense to link to the docs so people can RTFM.

Have added the following new jobs:

- **Group Cache Flush** - as that has been a long time recommendation.
- **Update Individual Postal Greeting** - required if this greeting is to be updated (which I had always assumed was done already)
- **Update Individual Addressee** - required if this greeting is to be updated
- **Dedupe Contacts** - as this is something that would be really useful to have set up from Day 1!

And re-named this job: Update Individual Email Greeting

**Comment by agileware-justin** [comment]
Test fail doesn't look related ( - test this please )

**Comment by seamuslee001** [comment]
I think this seems fine to me and noting this only affects new installs directly and I think that is fine flagging as merge ready in case anyone has any more ideas

**Comment by demeritcowboy** [comment]
Some people use the `name` in post-install scripts to enable/disable jobs so that might silently fail for the one with the name change, but I don't think it's a blocker.

Will need regen, but can be done after, just means the online demo here doesn't show the changes.

**Comment by demeritcowboy** [comment]
I've added a suggested note to the dev-digest about post-install scripts that will need updating.

And regen PR is at https://github.com/civicrm/civicrm-core/pull/28116

**Pull request merged**